pkg/image/apis/image/helper.go

```diff
@@ -397,13 +397,28 @@ func FollowTagReference(stream *ImageStream, tag string) (finalTag string, ref *
 			// no tag at the end of the rainbow
 			return tag, nil, false, multiple
 		}
-		if tagRef.From == nil || tagRef.From.Kind != "ImageStreamTag" || strings.Contains(tagRef.From.Name, ":") {
+		if tagRef.From == nil || tagRef.From.Kind != "ImageStreamTag" {
 			// terminating tag
 			return tag, &tagRef, true, multiple
 		}
 
-		// follow the referenec
-		tag = tagRef.From.Name
+		// The reference comes with three format patterns:
+		//   a) anotheris:sometag
```

Comment:
I don't think we allow imagestreamtags to be defined in terms of an imagestreamtag from another imagestream, @smarterclayton ?  (I'm basing that on this in the docs: 
```
Use a tracking tag for creating permanent aliases (for example, latest or stable). The tag works correctly only within a single image stream. Trying to create a cross-image-stream alias will produce an error.
```
https://docs.openshift.org/latest/dev_guide/managing_images.html#adding-tag

Though I do not see anything in our validation logic that enforces it)


If we do allow cross-imagestream aliasing, then this logic is incomplete, it should be following imagestream:tag reference until it resolves to a docker pull spec or an imagestreamimage.

Reply:
> I don't think we allow imagestreamtags to be defined in terms of an imagestreamtag from another imagestream,

I believe that it is correct (imagestreamtag from another is are not allowed). And the `oc tag` command has the validation.

https://github.com/openshift/origin/blob/607caee016274b5a352ee9739886cea9e497d87a/pkg/oc/cli/cmd/tag.go#L277-L280

However, the reason why I wrote `anotheris:sometag` handling in my PR is that `oc create` or `oc edit` can create the cross-imagestream aliasing.

Anyway, I will be waiting for @smarterclayton 's confirmation.

Reply:
yeah it seems like we have some inconsistency and it would be good to figure out what the actual intended behavior was.  Anyway either this function doesn't need the cross-imagestream handling (if it's not allowed) or it needs to be updated to actual resolve cross-imagestream references.  Either way I think the current behavior (returning the cross-imagestream reference) is wrong.

Reply:
Alright, I updated to stop returning cross-imagestream alias references. (Actually the update does not support cross-imagestream, but will change it if cross-imagestream alias is supported.).

Reply:
The `oc tag` is correct I've personally added it there, if we're missing that validation on the server side, one should be added.
The reason it was added to `oc tag` is that aliases should be updated when source is updated, but at the same time `oc tag` is meant to expose certain ISTags to users that normally might now have the access to the original. To prevent any of that trickery user is forced to create a permanent tag for cross-namespace cases.

Reply:
i've opened https://bugzilla.redhat.com/show_bug.cgi?id=1530340 to track what we should do about alerting users when they create invalid aliases.